### PR TITLE
Initial read-yaml action

### PR DESCRIPTION
**Pull request by kenodegard**
Per https://github.com/conda/infra/pull/508#discussion_r791616572

Implementing our own version of [KJ002/read-yaml](https://github.com/KJ002/read-yaml) to further simplify logic.

I have tested this manually on the command line so I'm about 75% sure this will work correctly.

### CODE REVIEW

File: read-yaml/action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    description: Path to YAML file in this repo to read.
+    required: true
+  scopes:
+    description: Path of keys to the value to extract from file (e.g. foo.bar).
+    required: false
+# cannot set outputs since they are dynamic based on the scopes value
+# outputs:
+#   data:
+#     description: Data read from YAML file.
+#     value: ${{ steps.read_yaml.outputs.data }}
```
Comment (jezdez) [inline read-yaml/action.yml:15]: Is this going to be a problem if the outputs aren't specified?

Comment (kenodegard) [reply read-yaml/action.yml:15]: AFAIK no, the outputs are optional and have no impact on what is actually outputted

see https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs

Comment (jezdez) [reply read-yaml/action.yml:15]: Awesome!

File: read-yaml/action.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      shell: bash -l {0}
+    - id: read_yaml
+      uses: actions/github-script@v5
+      with:
+        script: |
+          const yaml = require('js-yaml');
+
+          const path = core.getInput('path', { required: true });
+          const scopes = yaml.load(core.getInput('scopes') || ''))
+          const url = `/repos/${context.repo.owner}/${context.repo.repo}/contents/${path}`;
```
Comment (jezdez) [inline read-yaml/action.yml:32]: In the upstream action we just read from filesystem, was this not possible in this case?

Comment (kenodegard) [reply read-yaml/action.yml:32]: Yes, we could but then we'd have to clone the repo first, this eliminates doing the clone so its ever so slightly cheaper 🤷🏻‍♂️

Comment (jezdez) [reply read-yaml/action.yml:32]: Good point, thanks for elaborating!

**Review by jezdez — APPROVED** [review]
Super good, thank you!

**Pull request merged**